### PR TITLE
draft: cleanup low level loggers for plugin/library use - v4

### PR DESCRIPTION
**Pull request by jasonish**
Previous PR: https://github.com/OISF/suricata/pull/9493
Ticket: https://redmine.openinfosecfoundation.org/issues/7227

Changes from last PR:
- Cleanup and document packet logging
- Devguide entry

### CODE REVIEW

File: src/output-flow.c [hunk 1]
```diff
@@ -40,31 +40,33 @@ typedef struct OutputFlowLoggerThreadData_ {
...
 } OutputFlowLogger;
 
 static OutputFlowLogger *list = NULL;
 
-int OutputRegisterFlowLogger(const char *name, FlowLogger LogFunc,
-    OutputCtx *output_ctx, ThreadInitFunc ThreadInit,
-    ThreadDeinitFunc ThreadDeinit,
-    ThreadExitPrintStatsFunc ThreadExitPrintStats)
+int SCOutputRegisterFlowLogger(const char *name, FlowLogger LogFunc, void *initdata,
+        ThreadInitFunc ThreadInit, ThreadDeinitFunc ThreadDeinit)
```
Comment (jasonish) [inline src/output-flow.c:59]: `Output` and `Logger`? Might be nicer to be simply `SCRegisterFlowLogger`.

File: src/output-packet.c [hunk 2]
```diff
@@ -41,35 +41,30 @@ typedef struct OutputPacketLoggerThreadData_ {
...
 static OutputPacketLogger *list = NULL;
 
-int OutputRegisterPacketLogger(LoggerId logger_id, const char *name,
-    PacketLogger LogFunc, PacketLogCondition ConditionFunc,
-    OutputCtx *output_ctx, ThreadInitFunc ThreadInit,
-    ThreadDeinitFunc ThreadDeinit,
-    ThreadExitPrintStatsFunc ThreadExitPrintStats)
+int SCOutputRegisterPacketLogger(const char *name, PacketLogger LogFunc,
+        PacketLogCondition ConditionFunc, void *initdata, ThreadInitFunc ThreadInit,
+        ThreadDeinitFunc ThreadDeinit)
```
Comment (jasonish) [inline src/output-packet.c:56]: `SCRegisterPacketLogger`?

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 22272

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 22273

File: src/output-packet.h [hunk 3]
```diff
@@ -26,20 +26,44 @@
...
+ *     that passes the condition test.
+ * \param ConditionFunc A function to test if the packet should be passed to
+ *     the logging function.
+ * \param ThreadInitFunc Thread initialization function.
+ * \param ThreadDeinitFunc Thread de-initialization function.
+ *
+ * \retval 0 on success, -1 on failure.
+ */
+int SCOutputRegisterPacketLogger(LoggerId logger_id, const char *name, PacketLogger LogFunc,
+        PacketLogCondition ConditionFunc, void *initdata, ThreadInitFunc, ThreadDeinitFunc);
```
Comment (jasonish) [inline src/output-packet.h:61]: `SCRegisterPacketLogger`?

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 22281

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 22292

**Comment by jasonish** [comment]
Replaced by https://github.com/OISF/suricata/pull/11686.

**Pull request closed without merging**